client/my-sites/upgrades/cart/cart-plan-discount-ad.jsx

```diff
@@ -0,0 +1,80 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+import find from 'lodash/find';
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import CartAd from './cart-ad';
+import { cartItems } from 'lib/cart-values';
+import { fetchSitePlans } from 'state/sites/plans/actions';
+import { getPlansBySite } from 'state/sites/plans/selectors';
+import { isBusiness, isPlan, isPremium } from 'lib/products-values';
+import i18n from 'lib/mixins/i18n';
+import { shouldFetchSitePlans } from 'lib/plans';
+
+const CartPlanDiscountAd = React.createClass( {
+	propTypes: {
+		cart: React.PropTypes.object,
+		sitePlans: React.PropTypes.object
+	},
+
+	componentDidMount() {
+		this.props.fetchSitePlans( this.props.sitePlans, this.props.selectedSite );
+	},
+
+	render() {
+		const { cart, sitePlans } = this.props;
+		let plan;
+
+		if ( ! sitePlans.hasLoadedFromServer || ! cart.hasLoadedFromServer || ! cartItems.getAll( cart ).some( isPlan ) ) {
```

Comment:
`cartItems.getAll( cart ).some( isPlan )` could be replaced by `cartItems.hasPlan( cart )`